src/viewer.js

```diff
@@ -4008,49 +4012,93 @@ function doViewerResize(viewer, containerSize){
 }
 
 function handleArrowKeys(viewer) {
-    // Use the viewer's configured pan amount
-    const pixels = viewer.pixelsPerArrowPress || 40;
-    const panDelta = viewer.viewport.deltaPointsFromPixels(new OpenSeadragon.Point(pixels, pixels));
 
-    // Helper for key state (supports both object and Set)
-    function isDown(key) {
-        return viewer._keysDown && (viewer._keysDown[key] || (viewer._keysDown.has && viewer._keysDown.has(key)));
+    // Helper for key state
+    function isDown(code) {
+        return viewer._keysDown && viewer._keysDown[code];
     }
 
+     // Use the viewer's configured pan amount
+    const pixels = viewer.pixelsPerArrowPress;
+    const panDelta = viewer.viewport.deltaPointsFromPixels(new OpenSeadragon.Point(pixels, pixels));
+    const panDelta40 = viewer.viewport.deltaPointsFromPixels(new OpenSeadragon.Point(40, 40));
+
     // Shift key state
     const shift = isDown('Shift') || isDown('ShiftLeft') || isDown('ShiftRight');
 
-    // Up Arrow
-    if (!viewer.preventVerticalPan && isDown('ArrowUp')) {
-        if (shift) {
-            viewer.viewport.zoomBy(1.1);
-        } else {
-            viewer.viewport.panBy(new OpenSeadragon.Point(0, -panDelta.y));
+    // Up Arrow and Down Arrow
+    if (!viewer.preventVerticalPan) {
+        if (isDown('ArrowUp')) {
+            if (shift) {
+                viewer.viewport.zoomBy(1.1);
```

Comment:
Working on this I'm seeing we have a few bugs from before that you can fix. Having the `zoomBy` inside the `!viewer.preventVerticalPan` is a bug... the pan prevention shouldn't prevent zooming.